reaktive/src/jvmJsCommonMain/kotlin/com/badoo/reaktive/scheduler/BufferedExecutor.kt

```diff
@@ -1,18 +1,26 @@
 package com.badoo.reaktive.scheduler
 
+import com.badoo.reaktive.disposable.Disposable
 import com.badoo.reaktive.utils.queue.ArrayQueue
 import com.badoo.reaktive.utils.queue.Queue
 
 internal actual class BufferedExecutor<in T> actual constructor(
     private val executor: Scheduler.Executor,
     private val onNext: (T) -> Unit
-) {
+) : Disposable {
 
     private val monitor = Any()
     private val queue: Queue<T> = ArrayQueue()
     private var isDraining = false
     private val drainFunction = ::drain
 
+    override var isDisposed: Boolean = false
+        private set
+
+    override fun dispose() {
+        isDisposed = true
+    }
+
     actual fun submit(value: T) {
         synchronized(monitor) {
```

Comment:
The purpose of the `RefCounter` here is to protected the `Lock` instance from being used when it is already destroyed. Since the `Lock` is a resource in K/N, it requires explicit destroy. It is illegal to use the native `Lock` if its memory is reclaimed. There is no intention to protect the executor itself, it is just delegates calls to the consumer (where the protection should present).

Perhaps I should add a comment for `RefCounter` use case here?